tests/dispense.rs

```diff
@@ -174,42 +175,24 @@ async fn dispense_sends_coins_to_valid_address_hex_address() {
     let mut rng = StdRng::seed_from_u64(42);
     let recipient_address: Address = rng.gen();
 
-    _dispense_sends_coins_to_valid_address(
-        rng,
-        recipient_address.into(),
-        format!("{:#x}", &recipient_address),
-    )
-    .await
+    dispense_sends_coins_to_valid_address(rng, recipient_address.into()).await
 }
 
 #[tokio::test]
 async fn dispense_sends_coins_to_valid_address_non_hex() {
     let mut rng = StdRng::seed_from_u64(42);
     let recipient_address: Address = rng.gen();
 
-    _dispense_sends_coins_to_valid_address(
-        rng,
-        recipient_address.into(),
-        format!("{}", &recipient_address),
-    )
-    .await
+    dispense_sends_coins_to_valid_address(rng, recipient_address.into()).await
 }
 
-async fn _dispense_sends_coins_to_valid_address(
-    rng: StdRng,
-    recipient_address: Bech32Address,
-    recipient_address_str: String,
-) {
+async fn dispense_sends_coins_to_valid_address(rng: StdRng, recipient_address: Bech32Address) {
```

Comment:
Could you return the previous logic back, please? We used `recipient_address_str` to specify the hex and non-hex version of the address to test that we can process both.